pwndbg/commands/vmmap.py

```diff
@@ -17,27 +20,38 @@
 import pwndbg.vmmap
 
 
-@pwndbg.commands.QuietSloppyParsedCommand
+def map_type(s):
+    gdbval_or_str = pwndbg.commands.sloppy_gdb_parse(s)
+
+    # returns a module filter
+    if isinstance(gdbval_or_str, six.string_types):
+        module_name = gdbval_or_str
+        return lambda page: module_name in page.objfile
+
+    # returns an address filter
+    elif isinstance(gdbval_or_str, six.integer_types + (gdb.Value,)):
+        addr = gdbval_or_str
+        return lambda page: addr in page
+
+    else:
+        raise argparse.ArgumentTypeError('Unknown vmmap argument type.')
+
+parser = argparse.ArgumentParser()
+parser.description = 'Print the virtual memory map, or the specific mapping for the provided address / module name.'
+parser.add_argument('map', type=map_type, nargs='?', default=None,
+                    help='Address or module name.')
+
+
+@pwndbg.commands.ArgparsedCommand(parser)
 @pwndbg.commands.OnlyWhenRunning
 def vmmap(map=None):
-    """
```

Comment:
Shouldn't we do that for all `ArgparsedCommand`s?
I mean, we either should grab command description from docstring or specify it directly on `ArgparsedCommand`.